frontend/fingertips-frontend/components/forms/SearchForm/SearchForm.test.tsx

```diff
@@ -22,14 +23,48 @@ jest.mock('react', () => {
   };
 });
 
+const initialState: SearchFormState = {
+  indicator: '',
+  message: null,
+  errors: {},
+};
+
 test('snapshot test - renders the form', () => {
-  const container = render(<SearchForm indicator="" />);
+  const container = render(
+    registryWrapper(<SearchForm searchFormState={initialState} />)
+  );
 
   expect(container.asFragment()).toMatchSnapshot();
 });
 
 test('should have an input field to input the indicatorId', () => {
-  render(<SearchForm indicator="" />);
+  render(registryWrapper(<SearchForm searchFormState={initialState} />));
 
   expect(screen.getByTestId('search-form-input-indicator')).toBeInTheDocument();
 });
+
+test('should set the input field with indicator value from the form state', () => {
+  const indicatorState: SearchFormState = {
+    indicator: 'test value',
+    message: '',
+    errors: {},
+  };
+  render(registryWrapper(<SearchForm searchFormState={indicatorState} />));
+
+  expect(screen.getByDisplayValue('test value')).toHaveAttribute(
+    'id',
+    'indicator'
+  );
```

Comment:
Not a massive issue, but this seems a bit backwards. You're trying to assert the indicator input field has the expected value, but you're searching for elements with the expected display value and then checking the ID is that of the indicator field. Couldn't you just get the indicator field by ID (or the `search-form-input-indicator` `data-testid`) and then assert against its value?

Reply:
Yes I agree with this. 

Just something for us to be vary of. The data-testid `search-form-input-indicator` is against the full component. The input is within it so finding by data-testid and checking it's value doesn't quite work. Probably another reason to prefer using the `getByRole` instead.  
 
![Screenshot 2024-12-13 at 12 46 25](https://github.com/user-attachments/assets/a7ca6c0c-e06e-4620-8bf1-e488352b5604)